homeassistant/components/tcp.py

```diff
@@ -0,0 +1,125 @@
+"""
+homeassistant.components.tcp
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+A generic TCP socket component.
+"""
+import logging
+import socket
+import re
+from select import select
+
+from homeassistant.const import CONF_NAME, CONF_HOST
+from homeassistant.helpers.entity import Entity
+
+
+DOMAIN = "tcp"
+
+CONF_PORT = "port"
+CONF_TIMEOUT = "timeout"
+CONF_PAYLOAD = "payload"
+CONF_UNIT = "unit"
+CONF_VALUE_REGEX = "value_regex"
```

Comment:
Ah, inject the returned value as the `value` variable, I see. I'll implement that as `value_template`. Could I keep `value_regex` too though, or is anything you can do with regex possible with Jinja2?